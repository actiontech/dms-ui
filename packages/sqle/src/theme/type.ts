import { WorkflowRecordResV2StatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export type ExecWorkflowTheme = {
  list: {
    desc: {
      copyIconColor: string;
      hoverCopyIconBackgroundColor: string;
    };
  };
  create: {
    auditResult: {
      auditResultDrawer: {
        numberColor: string;
      };
      toggleButton: {
        borderColor: string;
        color: string;
        activeColor: string;
        bgColor: string;
        activeBgColor: string;
      };
      download: {
        borderColor: string;
        bgColor: string;
        boxShadow: string;
        itemColor: string;
        itemHoverColor: string;
        itemIconColor: string;
      };
    };
    form: {
      baseInfoTitleIconColor: string;
    };
    editForm: {
      titleColor: string;
      projectFlagIconColor: string;
    };
  };
  steps: {
    boxShadow: string;
  };
  common: {
    basicInfo: {
      borderColor: string;
      bgColor: string;
      titleColor: string;
      descColor: string;
    };
    auditResultFilter: {
      borderColor: string;
      bgColor: string;
      auditResultInfo: {
        itemBgColor: string;
        textColor: string;
        schemaValueColor: string;
      };
      options: {
        bgColor: string;
        activeBgColor: string;

        textColor: string;
        textActiveColor: string;
        numColor: string;
        numActiveColor: string;
      };
    };
  };
};

export type ReportStatisticsTheme = {
  bgColor: string;
  loadingColor: string;
  cardShow: {
    titleColor: string;
    iconColor: string;
    numberContColor: string;
    noteColor: string;
    contentFontSize: number;
    contentFontWeight: number;
  };
  CardWrapper: {
    titleColor: string;
    titleBorderColor: string;
  };
  TableTopList: {
    titleColor: string;
    noteTipColor: string;
    colorFillQuaternary: string;
    bgColor: {
      toColor: string;
      line1: string;
      line2: string;
      line3: string;
    };
  };
  ChartContTitle: {
    mainColor: string;
    mainSubColor: string;
    subContColor: string;
  };
  DatabaseTypeOrder: {
    default: {
      fontSize: number;
    };
    tooltip: {
      background: string;
      border: string;
      boxShadow: string;
    };
  };
  DatabaseSourceOrder: {
    default: {
      colorText: string;
      colorTextTertiary: string;
    };
  };
  LicenseStatistics: {
    ChartContTitle: {
      lineColor: string;
    };
    LicenseColumn: {
      defaultColor: string;
      fillColor: {
        column: string;
        columnBackground: string;
        xAxis: string;
        yAxis: string;
        state: {
          active: string;
        };
      };
    };
  };
  WorkOrderState: {
    ChartContTitle: {
      color: string;
      noteSubColor: string;
    };
  };
};

export type StatisticsTheme = {
  rectColor: string[];
  rectColorName: { [key: string]: string };
  auditRateStatus: {
    success: {
      color: string;
      bg: string;
    };
    warning: {
      color: string;
      bg: string;
    };
    error: {
      color: string;
      bg: string;
    };
    tip: {
      color: string;
      bg: string;
    };
  };
  auditResultStatusColor: {
    [key in WorkflowRecordResV2StatusEnum]: string;
  };
};

export type ProjectOverviewTheme = {
  SqlCount: {
    baseColor: string;
    grayColor: string;
  };
  DataSourceCount: {
    health: string;
    risk: string;
  };
  ProjectScore: {
    level: {
      dangerous: string;
      warning: string;
      good: string;
      excellent: string;
    };
    indicator: {
      pointer: string;
      pin: {
        fill: string;
        stroke: string;
      };
    };
  };
  ScanTask: {
    bar: {
      fill: string;
      bg: string;
      activeColor: string;
      label: {
        fill: string;
      };
      toolTip: {
        dotColor: string;
      };
    };
    detail: {
      stroke: string;
    };
  };
  DataSourcePerformance: {
    bar: {
      fill: string;
      bg: string;
    };
    toolTip: {
      dotColor: string;
    };
  };
};

export type WorkflowTemplateTheme = {
  progress: {
    normal: string;
    notice: string;
    warning: string;
    error: string;
    remainColor: string;
  };
  step: {
    honour: string;
    addFile: string;
    audit: string;
    sendPlane: string;
  };
  stepCard: {
    userAvatarBorder: string;
    boxShadow: string;
    hoverBoxShadow: string;
    backgroundColor: string;
    disableBorder: string;
    disabledBackgroundColor: string;
    disableHoverBackgroundColor: string;
    cardDesc: string;
    cardOperator: {
      borderTop: string;
      titleColor: string;
    };
    stepLineColor: string;
    stepReviewIndexTextColor: string;
    closeButtonBoxShadow: string;
    closeButtonFontSize: string;
    cardTitleFontWeight: number;
    descFontSize: string;
    operatorTitleFontWeight: number;
    stepReviewIndexTextFontWeight: number;
    authTextColor: string;
  };
  updateWorkflowTemplateStepInfo: {
    titleFontSize: string;
    titleFontWeight: number;
    infoFontSize: string;
    title: {
      titleWrapperBorderBottom: string;
      titleColor: string;
      titleInfoColor: string;
    };
    info: {
      titleColor: string;
    };
    stepInfoWrapper: {
      border: string;
      backgroundColor: string;
      textColor: string;
    };
    stepNodeAlert: {
      backgroundColor: string;
      alertTitleColor: string;
      alertContentColor: string;
    };
  };
  workflowTemplateAuthInfo: {
    topLevelFontWeight: number;
    authLevelFontWeight: number;
    authLevelFontSize: string;
    authInfoFontSize: string;
    borderBottom: string;
    authInfoColor: string;
  };
};

export type IconTheme = {
  execWorkFlow: {
    minusCircleFilledDisabled: string;
    profileSquareFilled: string;
    databaseFilled: string;
    schemaFilled: string;
    fileList: string;
    clock: string;
  };
  workflowTemplate: {
    common: string;
    userCircleFilled: string;
  };
};

export type DataSourceComparisonTheme = {
  comparisonResultDiffBackgroundColor: string;
  comparisonEntry: {
    card: {
      boxShadow: string;
      borderColor: string;
    };
    timeInfo: {
      color: string;
    };
    overview: {
      card: {
        hoverBoxShadow: string;
        suggestion: {
          color: string;
        };
        mode: {
          same: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          inconsistent: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          baseNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          comparisonNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
        };
      };
    };
    treeNode: {
      title: {
        bgColor: string;
      };
      node: {
        inconsistent: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        missing: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        new: {
          color: string;
          bgColor: string;
        };
        tag: {
          inconsistent: {
            color: string;
            bgColor: string;
          };
          missing: {
            color: string;
            bgColor: string;
          };
          new: {
            color: string;
            bgColor: string;
          };
        };
      };
    };
  };
};

export type KnowledgeTheme = {
  highlight: {
    color: string;
    background: string;
  };
  result: {
    list: {
      boxShadow: string;
      hoverBoxShadow: string;
    };
    description: {
      borderLeftColor: string;
    };
  };
  graph: {
    wrapper: {
      borderColor: string;
      backgroundColor: string;
      dotColor: string;
    };
    container: {
      fullScreenBgColor: string;
    };
    control: {
      borderColor: string;
    };
    edge: {
      color: string;
    };
  };
};
