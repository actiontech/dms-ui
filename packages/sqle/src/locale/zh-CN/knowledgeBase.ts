// eslint-disable-next-line import/no-anonymous-default-export
export default {
  pageTitle: '知识库',
  searchResults: {
    backToKnowledgeBase: '返回知识库',
    searchPlaceholder: '请输入关键字搜索你感兴趣的SQL语法'
  },
  search: {
    retrieveKnowledgeGraph: '检索知识图谱',
    selectedTags: '已选择标签'
  }
};
