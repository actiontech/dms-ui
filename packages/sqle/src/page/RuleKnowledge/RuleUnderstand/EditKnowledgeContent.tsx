import { EditKnowledgeContentProps } from './index.type';
import rehypeSanitize from 'rehype-sanitize';
import { RuleKnowledgeEditorStyleWrapper } from '../style';
import { useTranslation } from 'react-i18next';

const EditKnowledgeContent: React.FC<EditKnowledgeContentProps> = ({
  value,
  onChange,
  setHasDirtyData
}) => {
  const { t } = useTranslation();

  return (
    <RuleKnowledgeEditorStyleWrapper
      height="500px"
      previewOptions={{
        rehypePlugins: [[rehypeSanitize]]
      }}
      value={value}
      onChange={(v) => {
        onChange?.(v);
        setHasDirtyData(true);
      }}
      textareaProps={{
        placeholder: t('ruleKnowledge.editorPlaceholder')
      }}
    />
  );
};

export default EditKnowledgeContent;
