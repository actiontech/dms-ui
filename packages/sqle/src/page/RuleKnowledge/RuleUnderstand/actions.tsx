import { t } from '../../../locale';
import {
  PERMISSIONS,
  PermissionControl
} from '@actiontech/shared/lib/features';
import { ActionButton } from '@actiontech/shared';

export const RuleKnowledgeEditActions = (
  onEdit: () => void,
  loading: boolean
): Record<'edit-rule-knowledge', React.ReactNode> => ({
  'edit-rule-knowledge': (
    <PermissionControl
      permission={PERMISSIONS.ACTIONS.SQLE.RULE_KNOWLEDGE.EDIT}
    >
      <ActionButton
        type="primary"
        text={t('ruleKnowledge.edit')}
        onClick={onEdit}
        disabled={loading}
      />
    </PermissionControl>
  )
});
