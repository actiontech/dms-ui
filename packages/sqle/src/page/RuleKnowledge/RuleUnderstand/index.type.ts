import { MDEditorProps } from '@uiw/react-md-editor';

export type RuleUnderstandProps = {
  content?: string;
  loading: boolean;
  isModifying: boolean;
  editValue: MDEditorProps['value'];
  setEditValue: (val: MDEditorProps['value']) => void;
  setHasDirtyData: (val: boolean) => void;
  ruleName: string;
};

export type EditKnowledgeContentProps = {
  value: MDEditorProps['value'];
  onChange: MDEditorProps['onChange'];
  setHasDirtyData: (val: boolean) => void;
};
