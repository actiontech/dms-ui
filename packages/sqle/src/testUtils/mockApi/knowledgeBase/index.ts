import knowledge_base from '@actiontech/shared/lib/api/sqle/service/knowledge_base';
import {
  MockSpyApy,
  createSpySuccessResponse
} from '@actiontech/shared/lib/testUtil/mockApi';
import {
  mockKnowledgeBaseTagListData,
  mockKnowledgeBaseListData,
  mockKnowledgeGraphData
} from './data';

class KnowledgeBase implements MockSpyApy {
  public mockAllApi(): void {
    this.getKnowledgeBaseTagList();
    this.getKnowledgeBaseList();
    this.getKnowledgeGraph();
  }

  public getKnowledgeBaseTagList() {
    const spy = jest.spyOn(knowledge_base, 'getKnowledgeBaseTagList');
    spy.mockImplementation(() =>
      createSpySuccessResponse({
        data: mockKnowledgeBaseTagListData
      })
    );
    return spy;
  }

  public getKnowledgeBaseList() {
    const spy = jest.spyOn(knowledge_base, 'getKnowledgeBaseList');
    spy.mockImplementation(() =>
      createSpySuccessResponse({
        data: mockKnowledgeBaseListData,
        total_nums: 30
      })
    );
    return spy;
  }

  public getKnowledgeGraph() {
    const spy = jest.spyOn(knowledge_base, 'getKnowledgeGraph');
    spy.mockImplementation(() =>
      createSpySuccessResponse({
        data: mockKnowledgeGraphData
      })
    );
    return spy;
  }
}

export default new KnowledgeBase();
