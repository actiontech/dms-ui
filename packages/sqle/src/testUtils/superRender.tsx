import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
