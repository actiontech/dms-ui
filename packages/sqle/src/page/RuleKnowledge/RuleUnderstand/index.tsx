import {
  Card,
  Empty,
  Space,
  Typography,
  Spin,
  message,
  Popconfirm
} from 'antd';
import { RuleUnderstandProps } from './index.type';
import { useTranslation } from 'react-i18next';
import { EmptyBox, BasicButton } from '@actiontech/shared';
import { useBoolean } from 'ahooks';
import EditKnowledgeContent from './EditKnowledgeContent';
import { useEffect, useState } from 'react';
import rehypeSanitize from 'rehype-sanitize';
import rule_template from '@actiontech/shared/lib/api/sqle/service/rule_template';
import { ResponseCode } from '../../../data/common';
import { RuleKnowledgeMarkDownStyleWrapper } from '../style';
import classNames from 'classnames';
import { RuleKnowledgeEditActions } from './actions';

const RuleUnderstand: React.FC<RuleUnderstandProps> = ({
  content,
  ruleName,
  refresh,
  dbType,
  loading,
  isCustomRule
}) => {
  const { t } = useTranslation();
  const [modifyFlag, { setTrue: startModify, setFalse: modifyFinish }] =
    useBoolean();

  const [hasDirtyData, setHasDirtyData] = useState(false);

  const [editValue, setEditValue] = useState<string>();

  const [submitLoading, { setFalse: submitFinish, setTrue: startSubmit }] =
    useBoolean();

  const [messageApi, contextHolder] = message.useMessage();

  const submit = () => {
    startSubmit();
    const request = isCustomRule
      ? rule_template.updateCustomRuleKnowledge({
          rule_name: ruleName,
          knowledge_content: editValue,
          db_type: dbType
        })
      : rule_template.updateRuleKnowledge({
          rule_name: ruleName,
          knowledge_content: editValue,
          db_type: dbType
        });

    request
      .then((res) => {
        if (res.data.code === ResponseCode.SUCCESS) {
          messageApi.success(t('ruleKnowledge.successTips'));
          setHasDirtyData(false);
          modifyFinish();
          refresh();
          setEditValue('');
        }
      })
      .finally(() => {
        submitFinish();
      });
  };

  useEffect(() => {
    if (modifyFlag) {
      setEditValue(content);
    }
  }, [content, modifyFlag]);

  const actions = RuleKnowledgeEditActions(startModify, loading);

  return (
    <>
      {contextHolder}
      <Card
        title={t('ruleKnowledge.ruleUnderstanding')}
        extra={
          !modifyFlag ? (
            actions['edit-rule-knowledge']
          ) : (
            <Space>
              <EmptyBox
                if={hasDirtyData}
                defaultNode={
                  <BasicButton disabled={submitLoading} onClick={modifyFinish}>
                    {t('common.cancel')}
                  </BasicButton>
                }
              >
                <Popconfirm
                  title={t('ruleKnowledge.hasDirtyDataTips')}
                  okText={t('common.ok')}
                  cancelText={t('common.cancel')}
                  onConfirm={() => {
                    modifyFinish();
                    setHasDirtyData(false);
                    setEditValue(content);
                  }}
                >
                  <BasicButton disabled={submitLoading}>
                    {t('common.cancel')}
                  </BasicButton>
                </Popconfirm>
              </EmptyBox>
              <BasicButton
                disabled={submitLoading}
                type="primary"
                onClick={submit}
              >
                {t('common.submit')}
              </BasicButton>
            </Space>
          )
        }
        className={classNames({ 'card-editing-status': modifyFlag })}
      >
        <Spin spinning={loading}>
          {modifyFlag ? (
            <EditKnowledgeContent
              value={editValue}
              onChange={setEditValue}
              setHasDirtyData={setHasDirtyData}
            />
          ) : (
            <EmptyBox
              if={!!content}
              defaultNode={
                <Empty
                  image={Empty.PRESENTED_IMAGE_SIMPLE}
                  description={
                    <Typography.Text type="secondary">
                      {t('ruleKnowledge.noData')}
                    </Typography.Text>
                  }
                />
              }
            >
              <RuleKnowledgeMarkDownStyleWrapper
                source={content}
                rehypePlugins={[rehypeSanitize]}
              />
            </EmptyBox>
          )}
        </Spin>
      </Card>
    </>
  );
};

export default RuleUnderstand;
