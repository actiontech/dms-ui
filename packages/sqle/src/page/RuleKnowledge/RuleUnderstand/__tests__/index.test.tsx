import RuleUnderstand from '..';
import { sqleSuperRender } from '../../../../testUtils/superRender';
import {
  ruleNameFirst as ruleName,
  mockMarkdownWithCustomCodeBlock
} from '../../../../testUtils/mockApi/rule_template/data';
import { RuleUnderstandProps } from '../index.type';
import { screen, act } from '@testing-library/react';
import {
  ignoreConsoleErrors,
  UtilsConsoleErrorStringsEnum
} from '@actiontech/shared/lib/testUtil/common';
import knowledgeBase from '../../../../testUtils/mockApi/knowledgeBase';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn()
  };
});

jest.mock('rehype-rewrite', () => {
  return {
    getCodeString: jest.fn()
  };
});

describe('page/RuleUnderstand', () => {
  let getKnowledgeGraphSpy: jest.SpyInstance;
  ignoreConsoleErrors([UtilsConsoleErrorStringsEnum.INVALID_CUSTOM_ATTRIBUTE]);

  const ruleUnderstandProps: RuleUnderstandProps = {
    ruleName,
    content: mockMarkdownWithCustomCodeBlock,
    loading: false,
    isModifying: false,
    editValue: '',
    setEditValue: jest.fn(),
    setHasDirtyData: jest.fn()
  };

  beforeEach(() => {
    jest.useFakeTimers();
    getKnowledgeGraphSpy = knowledgeBase.getKnowledgeGraph();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  const customRender = (data: RuleUnderstandProps) => {
    return sqleSuperRender(<RuleUnderstand {...data} />);
  };

  it('render empty data', () => {
    const { baseElement } = customRender({
      ...ruleUnderstandProps,
      content: undefined
    });
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText('暂无数据, 请编辑')).toBeInTheDocument();
  });

  it('render content data', async () => {
    const { baseElement } = customRender({ ...ruleUnderstandProps });
    await act(async () => jest.advanceTimersByTime(3000));
    expect(getKnowledgeGraphSpy).toHaveBeenCalledTimes(1);
    expect(getKnowledgeGraphSpy).toHaveBeenCalledWith({
      filter_by_rule_name: ruleName
    });
    expect(baseElement).toMatchSnapshot();
  });
});
