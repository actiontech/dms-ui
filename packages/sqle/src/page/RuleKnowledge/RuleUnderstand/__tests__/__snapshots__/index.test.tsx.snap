// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page/RuleUnderstand render content data 1`] = `
<body>
  <div>
    <div
      class=" css-1yb442y"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="markdown-container"
          >
            <div
              class="css-1pkzawh"
              components="[object Object]"
              rehypeplugins="() => jest.fn()"
            >
              
  ## 标签
  \`\`\`label [数据库版本:MySQL 5.7.25,检查方式:不连库审核]\`\`\`
  ### 示例
  \`\`\`sql
  select * from customers where name is not null;        
  \`\`\`
  ### SQL Diff
  \`\`\`sql_diff
    ---before
      SELECT * FROM Users
      WHERE id &gt; 0 
      AND status = 'active'
    ---after
      SELECT id, name 
      FROM Users
      WHERE id &gt; 0 
      AND status = 'active'
  \`\`\`

            </div>
            <h2
              class="ant-typography graph-title css-dev-only-do-not-override-txh9fw"
              id="header-0"
            >
              关联的语法节点
            </h2>
            <div
              class="css-1c94q4z"
            >
              <div
                class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
              >
                <div>
                  <div
                    aria-busy="true"
                    aria-live="polite"
                    class="ant-spin ant-spin-spinning ant-spin-show-text css-dev-only-do-not-override-txh9fw"
                  >
                    <span
                      class="ant-spin-dot ant-spin-dot-spin"
                    >
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                      <i
                        class="ant-spin-dot-item"
                      />
                    </span>
                    <div
                      class="ant-spin-text"
                    >
                      加载知识图谱中...
                    </div>
                  </div>
                </div>
                <div
                  class="ant-spin-container ant-spin-blur"
                />
              </div>
            </div>
          </div>
          <div>
            <div
              class=""
            >
              <div
                class="css-dev-only-do-not-override-txh9fw ant-anchor-wrapper"
                style="max-height: 100vh;"
              >
                <div
                  class="ant-anchor"
                >
                  <span
                    class="ant-anchor-ink ant-anchor-ink-visible"
                    style="top: 0px; height: 0px;"
                  />
                  <div
                    class="ant-anchor-link ant-anchor-link-active"
                  >
                    <a
                      class="ant-anchor-link-title ant-anchor-link-title-active"
                      href="#header-0"
                      title="关联的语法节点"
                    >
                      关联的语法节点
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`page/RuleUnderstand render empty data 1`] = `
<body>
  <div>
    <div
      class=" css-1yb442y"
    >
      <div
        class="ant-spin-nested-loading css-dev-only-do-not-override-txh9fw"
      >
        <div>
          <div
            aria-busy="true"
            aria-live="polite"
            class="ant-spin ant-spin-spinning css-dev-only-do-not-override-txh9fw"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
        <div
          class="ant-spin-container ant-spin-blur"
        >
          <div
            class="css-dev-only-do-not-override-txh9fw ant-empty ant-empty-normal"
          >
            <div
              class="ant-empty-image"
            >
              <svg
                height="41"
                viewBox="0 0 64 41"
                width="64"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  transform="translate(0 1)"
                >
                  <ellipse
                    cx="32"
                    cy="33"
                    fill="#f5f5f5"
                    rx="32"
                    ry="7"
                  />
                  <g
                    fill-rule="nonzero"
                    stroke="#d9d9d9"
                  >
                    <path
                      d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                    />
                    <path
                      d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                      fill="#fafafa"
                    />
                  </g>
                </g>
              </svg>
            </div>
            <div
              class="ant-empty-description"
            >
              <span
                class="ant-typography ant-typography-secondary css-dev-only-do-not-override-txh9fw"
              >
                暂无数据, 请编辑
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
